Allow verbose/quiet level to be specified via config file and env var (#8578)
pypa/pip: docs/html/user_guide.rst

    no-compile = no
    no-warn-script-location = false

For options which can be repeated like ``--verbose`` and ``--quite``,

> a nonnegative integer could be used to represent the level to be specified:

Reviewer: ```suggestion
a non-negative integer can be used to represent the level to be specified:
```

Reviewer: `nonnegative` -> `non-negative` can be a personal choice. I just think the latter is more readable.

format ``PIP_<UPPER_LONG_NAME>`` . Dashes (``-``) have to be replaced with
underscores (``_``).

For example, setting ``PIP_DEFAULT_TIMEOUT`` to 60 has the same effect as

> passing the option to pip directly::

Reviewer 2: I feel the previous rendering (using code blocks) is more readable than prose form.

Author: I have the same feeling too, but I tried to avoid variable setting syntax because I couldn't find anything that is portable, e.g. [PowerShell uses the hideous `$Env:<variable-name> = "<new-value>"`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables).

Reviewer 2: You can use `.. code-block:: none` to have a code block without syntax highlighting.

Author: > Maybe it would help to demonstrate this in Python?

Probably, but I'm not sure how to phrase the surrounding text to interject `os.environ['PIP_FOO'] = bar` that would fit into the comparison with `pip --foo=bar ...`.  I think waiting for https://github.com/pypa/pip/pull/8589#issuecomment-662054242 might be a better idea, or doing a follow-up after it gets merged, depending on which PR will get merged first.

Reviewer 3: Honestly, I've never heard a Windows user complain about being confused by

```
export PIP_DEFAULT_TIMEOUT=60
```

I don't think the issue is important enough to warrant changing things, let's just stick with what we've been using up to now.

Author: As somebody who doesn't use Windows, I am delighted to revert back to using `export` :smile:

    pip --default-timeout=60 [...]

For command line options which can be repeated, if they do not take any argument

> like ``--quite`` or ``--verbose``, use a nonnegative integer to represent

Reviewer: ```suggestion
like ``--quite`` or ``--verbose``, use a non-negative integer to represent
```

> the number of repeatition, for instance setting ``PIP_VERBOSE`` to 3 is

Reviewer: ```suggestion
the number of repetition, for instance, setting ``PIP_VERBOSE`` to 3 is
```

> equivalent to calling pip with ``-vvv``.

Reviewer 2: I feel the previous rendering (using code blocks) is more readable than prose form.

> Fruitful repeatable options however, can be specified via environment variable

Reviewer: ```suggestion
Fruitful repeatable options, however, can be specified via environment variable
```

> as a space-separated list.  For example having ``PIP_FIND_LINKS`` set to

Reviewer: ```suggestion
as a space-separated list. For example, having ``PIP_FIND_LINKS`` set to
```

``"http://mirror1.example.com http://mirror2.example.com"`` is the same as
calling ::

> 

Reviewer 3: Again, let's leave the existing text as it is, and just add the extra text about options that don't take a value but can be repeated. (If we do want to change the text, though, "Fruitful repeatable options" doesn't mean anything to me - maybe it's a phrase that didn't translate well to English? Or maybe it's just something I've never heard before?)

Author: *Fruitful function* is used in [a book I read quite some time ago](https://www.greenteapress.com/thinkpython/thinkCSpy/html/chap05.html) to refer to function with a positive number of parameters but after looking it up all results point to this exact book (or its derivatives), so I think it's definitely not something to be used in a documentation and I am sorry for not checking it more carefully before hand<sup>*nervously thinking about everything I've written in the last few years*</sup>

What is your suggestion in this case to document the `count` and `append` options.  Those don't seem to be intuitive out of implementation context, and originally `count` options was not documented.

Reviewer 3: >  I am sorry for not checking it more carefully before hand

Not a problem!

> What is your suggestion in this case

I would go with something like:

```suggestion

For command line options which can be repeated, use a space to separate
multiple values. For example::

    export PIP_FIND_LINKS="http://mirror1.example.com http://mirror2.example.com"

is the same as calling::

        pip install --find-links=http://mirror1.example.com --find-links=http://mirror2.example.com
        
Options that do not take a value, but can be repeated (such as ``--verbose``) can be specified using the number of repetitions, so::

    export PIP_VERBOSE=3

is the same as calling::

    pip install -vvv
```

I'm not sure if github's "Suggestion" thing handled this correctly, but I hope you see what I mean, leave the existing text alone and just add a new paragraph about count-style options.

Author: GitHub suggestion doesn't seem to work, but this is clearer than what I wrote earlier so I'll apply it manually.
